cluster/gce/gci/configure-helper.sh

```diff
@@ -66,11 +68,74 @@ function create-dirs {
   fi
 }
 
-# Formats the given device ($1) if needed and mounts it at given mount point
+# Gets the total number of $(1) and $(2) type disks specified
+# by the user in ${NODE_LOCAL_SSDS_EXT}
+function get-local-disk-num() {
+  local interface=$1
+  local format=$2
+
+  localdisknum=0
+  if [[ ! -z ${NODE_LOCAL_SSDS_EXT} ]]; then
+    IFS=";" read -r -a ssdgroups <<< ${NODE_LOCAL_SSDS_ERXT}
+    for ssdgroup in "${ssdgroups[@]}"
+    do
+      IFS="," read -r -a ssdopts <<< ${ssdgroup}
+      local opnum=${ssdopts[0]}
+      local opinterface=${ssdopts[1]}
+      local opformat=${ssdopts[2]}
+
+      if [[ ${opformat,,} == ${format,,} && ${opinterface,,} == ${interface,,} ]]; then
+        localdisknum=$((localdisknum+opnum))
+      fi
+    done
+  fi
+}
+
+# Creates a symlink for a ($1) so that it may be used as block storage
+function safe-block-symlink(){
+  local device=$1
+  local interface=$2
+  
+  mkdir -p "${UUID_MNT_PREFIX}-${interface}-block"
+  ssdmap="/home/kubernetes/localssdmap.txt"
+  echo "Creating symlink for block devices..."
+
+  if [[ ! -e ${ssdmap} ]]; then
+    touch ${ssdmap}
+    chmod +w ${ssdmap}
+  fi
+
+  # each line of the ssdmap looks like "${device} persistent-uuid"
+  if [[ ! -z $(grep ${device} ${ssdmap}) ]]; then
+    #create symlink based on saved uuid
+    local myuuid=$(grep ${device} ${ssdmap} | cut -d ' ' -f 2)
+  else
+    # generate new uuid and add it to the map
+    local myuuid=$(uuidgen)
+    if [[ ! ${?} -eq 0 ]]; then
+      echo "Failed to generate valid UUID with uuidgen" >&2
+      exit 2
+    fi
+    echo "${device} ${myuuid}" >> ${ssdmap}
+  fi
+
+  if [[ -z ${myuuid} ]]; then
+    echo "Failed to get a uuid for device ${device} when symlinking." >&2
+    exit 2
+  fi
+
+  local sym="${UUID_MNT_PREFIX}-${interface}-block/local-ssd-${myuuid}"
```

Comment:
Now that the uuid is persisted, can we put the symlink under /dev/disk/by-id/ instead of /mnt/disks?  To avoid some confusion because /mnt is generally associated with mount points and not raw devices.